x-pack/plugins/cloud_security_posture/server/lib/check_index_status.ts

```diff
@@ -13,21 +13,42 @@ export const checkIndexStatus = async (
   esClient: ElasticsearchClient,
   index: string,
   logger: Logger,
-  postureType?: PostureTypes
+  postureType?: PostureTypes,
+  retentionTime?: number
 ): Promise<IndexStatus> => {
   const query =
     !postureType || postureType === 'all' || postureType === 'vuln_mgmt'
-      ? undefined
-      : {
+      ? {
           bool: {
-            filter: {
-              term: {
-                safe_posture_type: postureType,
+            must: {
+              range: {
+                '@timestamp': {
+                  gte: `now-${retentionTime}h/h`,
+                  lte: 'now/h',
+                },
               },
             },
           },
+        }
+      : {
+          bool: {
+            filter: [
+              {
+                term: {
+                  safe_posture_type: postureType,
```

Comment:
also as the difference is only in the addition of 
```
 filter: {
              term: {
                safe_posture_type: postureType,
              },
```
let's make this part added based on the condition. Right now it's hard to understand from the first glance that the `range` part is identical and only `safe_posture_type` is different